bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/SortedLedgerStorage.java

```diff
@@ -260,7 +261,8 @@ public void cancelWaitForLastAddConfirmedUpdate(long ledgerId,
     @Override
     public void checkpoint(final Checkpoint checkpoint) throws IOException {
         long numBytesFlushed = memTable.flush(this, checkpoint);
-        interleavedLedgerStorage.getEntryLogger().prepareSortedLedgerStorageCheckpoint(numBytesFlushed);
+        ((DefaultEntryLogger) interleavedLedgerStorage.getEntryLogger())
```

Comment:
This looks like a problem.it means that this implementation works only with the default implementation of EntryLogger

We should at least fail fast at construction time and do not wait for this method to be called.

Reply:
We are using `DefaultEntryLogger` to initiate InterleavedLedgerStorage entryLogger instance. So we can convert it into `DefaultEntryLogger` here.